crates/dbs-virtio-devices/src/vsock/backend/unix_stream.rs

```diff
@@ -2,15 +2,74 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use std::any::Any;
+use std::io::{Read, Write};
 use std::os::unix::io::{AsRawFd, RawFd};
 use std::os::unix::net::{UnixListener, UnixStream};
 use std::time::Duration;
 
 use log::info;
+use sendfd::RecvWithFd;
 
 use super::super::{Result, VsockError};
 use super::{VsockBackend, VsockBackendType, VsockStream};
 
+pub struct HybridUnixStream {
+    pub unix_stream: Box<dyn VsockStream>,
+    pub slave_stream: Option<Box<dyn VsockStream>>,
+}
+
+impl VsockStream for HybridUnixStream {
+    fn backend_type(&self) -> VsockBackendType {
+        self.unix_stream.backend_type()
+    }
+
+    fn set_nonblocking(&mut self, nonblocking: bool) -> std::io::Result<()> {
+        self.unix_stream.set_nonblocking(nonblocking)
+    }
+
+    fn set_read_timeout(&mut self, dur: Option<Duration>) -> std::io::Result<()> {
+        self.unix_stream.set_read_timeout(dur)
+    }
+
+    fn set_write_timeout(&mut self, dur: Option<Duration>) -> std::io::Result<()> {
+        self.unix_stream.set_write_timeout(dur)
+    }
+
+    fn as_any(&self) -> &dyn Any {
+        self.unix_stream.as_any()
+    }
+
+    fn recv_data_fd(&self, bytes: &mut [u8], fds: &mut [RawFd]) -> std::io::Result<(usize, usize)> {
+        self.unix_stream.recv_data_fd(bytes, fds)
+    }
+}
+
+impl AsRawFd for HybridUnixStream {
+    fn as_raw_fd(&self) -> RawFd {
+        self.unix_stream.as_raw_fd()
+    }
+}
+
+impl Read for HybridUnixStream {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        self.unix_stream.read(buf)
+    }
+}
+
+impl Write for HybridUnixStream {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        if let Some(mut stream) = self.slave_stream.take() {
```

Comment:
Yes, the slave stream only used for reply "ok <port>",  and the main stream is used for data transportation.  Why we use the slave stream to answer the "ok <port>"? Because the slave stream connection can be hold by the caller, but the main stream connection may be not controlled by the caller and would only be used send/receive data.

Reply:
could you add a comment here?